### ISSUE TITLE
Return raw bytes from endpoint function and preprocess it before send

### ISSUE THREAD

**Issue by Redict** [tiangolo/fastapi#5668]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastApi

class RawResponse(Response):
    media_type = "binary/octet-stream"
     
    def render(self, content: bytes) -> bytes:
        self.body = bytes([b ^ 0x54 for b in content])

app = FastApi(default_response_class=RawResponse)

@app.get("/")
def endpoint_fn(_: Request):
    return b"\xf6j,\x90\xae\\\xf3\xd3\xaa"
```

### Description

For this code I get `'utf-8' codec can't decode byte` at routing.py:234 (`serialize_response` call). How can I override this? 
Only one hackish solution that I've found is to b64encode(bytes) and in `RawResponse` render b64decode(content) then xor it

### Operating System

Windows

### FastAPI Version

0.68.1

### Python Version

Python 3.9.2

**Comment by ThirVondukr** [comment]
You can return Response (or RawResponse) object directly so fastapi doesn't try to serialize it:
```py
class RawResponse(Response):
    media_type = "binary/octet-stream"
    def render(self, content: bytes) -> bytes:
        return bytes([b ^ 0x54 for b in content])

app = FastAPI()

@app.get("/test")
async def test():
    return RawResponse(content=b"\xf6j,\x90\xae\\\xf3\xd3\xaa")
```

**Comment by Redict** [comment]
> You can return Response (or RawResponse) object directly so fastapi doesn't try to serialize it:
> 
> ```python
> class RawResponse(Response):
>     media_type = "binary/octet-stream"
> 
>     def render(self, content: bytes) -> bytes:
>         return bytes([b ^ 0x54 for b in content])
> 
> app = FastAPI()
> 
> @app.get("/test")
> async def test():
>     return RawResponse(content=b"\xf6j,\x90\xae\\\xf3\xd3\xaa")
> ```

I need to use this in a hundred endpoints, so it doesn't work for me because it's too redundant

**Comment by ThirVondukr** [comment]
I came up with this solution that wraps endpoint function to return `RawResponse` automatically.  
I think it's worth to leave this issue open since someone might have a better solution, and I personally don't really my solution that much 🤔 
```py
def wrap_response(endpoint: Any, response_cls: type[Response]) -> Any:
    @functools.wraps(endpoint)
    async def wrapper(*args, **kwargs):
        response = await endpoint(*args, **kwargs)
        return response_cls(content=response)

    return wrapper


class RawResponse(Response):
    media_type = "binary/octet-stream"

    def render(self, content: bytes) -> bytes:
        return bytes([b ^ 0x54 for b in content])


class RawRoute(APIRoute):
    def get_route_handler(self) -> Callable:
        self.dependant.call = wrap_response(endpoint=self.dependant.call, response_cls=RawResponse)
        return super().get_route_handler()
```

**Comment by TheJumpyWizard** [comment]
Any luck with implementing the above? @Redict

**Comment by ThirVondukr** [comment]
> Any luck with implementing the above? @Redict

Yep, it works, but I wonder if there's a better way of doing that, so maybe someone else would get to this issue too

**Comment by yinziyan1206** [comment]
![image](https://user-images.githubusercontent.com/37829370/203504466-504d54d5-5823-4c62-8ca4-01b4564fd4a0.png)

**Comment by ThirVondukr** [comment]
@yinziyan1206 That doesn't solve the problem really 🤔 
Main issue is that fastapi tries to `decode` bytes with utf-8 encoding before creating response class

**Comment by yinziyan1206** [comment]
Oops. you just return `bytes` and `json_encoder` use `lambda o: o.decode()` for `bytes`. Maybe you should change the default encoding type or overwrite the `ENCODERS_BY_TYPE` in pydantic, but i suggest `decode` it by yourself

**Comment by ThirVondukr** [comment]
They need to return binary data, and overriding pydantic internals doesn't seem like a good idea 👀 
So using decorator or something similar seems like the best way of doing that
